charms/cephfs-server-proxy/terraform/main.tf

```diff
@@ -0,0 +1,29 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "juju_application" "cephfs-server-proxy" {
+  name  = var.app_name
+  model = var.model_name
+
+  charm {
+    name     = "cephfs-server-proxy"
+    base     = var.base
+    channel  = var.channel
+    revision = var.revision
+  }
+
+  config      = var.config
+  constraints = var.constraints
+  units       = 1
```

Comment:
You'd want this field to be equal to `var.units`:

```suggestion
  units       = var.units
```

By hardcoding the the value of `1` above, this means that even if `units` is presented as a tuneable to the average Terraform enjoyer, they will only ever have 1 unit deployed since the value is hardcoded the `juju_application` resource being declared here.